azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -15,34 +15,405 @@
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_SERVICE_NAME_PREFIX,
+            DEFAULT_SERVICE_NAME_PREFIX);
+    this.clusterName = azkProps.getString(ConfigurationKeys.AZKABAN_CLUSTER_NAME,
+        DEFAULT_CLUSTER_NAME);
+    this.cpuLimit = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_CPU_LIMIT,
+            CPU_LIMIT);
+    this.cpuRequest = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_CPU_REQUEST,
+            DEFAULT_CPU_REQUEST);
+    this.memoryLimit = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_MEMORY_LIMIT,
+            MEMORY_LIMIT);
+    this.memoryRequest = azkProps
+        .getString(ContainerizedExecutionManagerProperties.KUBERNETES_FLOW_CONTAINER_MEMORY_REQUEST,
+            DEFAULT_MEMORY_REQUEST);
+
+    try {
+      // Path to the configuration file for Kubernetes which contains information about
+      // Kubernetes API Server and identity for authentication
+      String kubeConfigPath = azkProps
+          .getString(ContainerizedExecutionManagerProperties.KUBERNETES_KUBE_CONFIG_PATH);
+      logger.info("Kube config path is : {}", kubeConfigPath);
+      this.client =
+          ClientBuilder.kubeconfig(KubeConfig.loadKubeConfig(
+              Files.newBufferedReader(Paths.get(kubeConfigPath), Charset.defaultCharset())))
+              .build();
+      this.coreV1Api = new CoreV1Api(client);
+    } catch (IOException exception) {
+      logger.error("Unable to read kube config file: {}", exception.getMessage());
+      throw new ExecutorManagerException(exception);
+    }
   }
 
+  /**
+   * This method is used to create container during dispatch of execution. It will create pod for a
+   * flow execution. It will also create a service for a pod if azkaban.kubernetes.service .required
+   * property is set.
+   *
+   * @param executionId
+   * @throws ExecutorManagerException
+   */
   @Override
   public void createContainer(final int executionId) throws ExecutorManagerException {
-    logger.info("Container is created");
+    createPod(executionId);
+    if (isServiceRequired()) {
+      createService(executionId);
+    }
   }
 
+  /**
+   * This method is used to delete container. It will delete pod for a flow execution. If the
+   * service was created then it will also delete the service. This method can be called as a part
+   * of cleanup process for containers in case containers didn't shutdown gracefully.
+   *
+   * @param executionId
+   * @throws ExecutorManagerException
+   */
   @Override
   public void deleteContainer(final int executionId) throws ExecutorManagerException {
-    logger.info("Container is deleted");
+    deletePod(executionId);
+    if (isServiceRequired()) {
+      deleteService(executionId);
+    }
+  }
+
+  /**
+   * This method is used to create pod. 1. Fetch jobTypes for the flow 2. Fetch flow parameters for
```

Comment:
Please start all the numbered steps from a new line for better readability.

Reply:
Thanks for pointing it out. Done.